crates/xline/src/storage/lease_store/mod.rs

```diff
@@ -98,18 +101,26 @@ impl LeaseStore {
     }
 
     /// sync a lease request
-    pub(crate) fn after_sync(
+    pub(crate) fn after_sync<T, I>(
         &self,
         request: &RequestWrapper,
         revision_gen: &RevisionNumberGeneratorState<'_>,
-    ) -> Result<(SyncResponse, Vec<WriteOp>), ExecuteError> {
-        let revision = if request.skip_lease_revision() {
-            revision_gen.get()
-        } else {
+        txn_db: &T,
+        index: &I,
+    ) -> Result<(SyncResponse, Vec<WriteOp>), ExecuteError>
+    where
+        T: XlineStorageOps + TransactionApi,
+        I: IndexOperate,
+    {
+        let next_revision = revision_gen.get().overflow_add(1);
+        let updated = self.sync_request(request, next_revision, txn_db, index)?;
+        let rev = if updated {
             revision_gen.next()
+        } else {
+            revision_gen.get()
         };
-        self.sync_request(request, revision)
-            .map(|(rev, ops)| (SyncResponse::new(rev), ops))
+        // TODO: return only a `SyncResponse`
```

Comment:
Do we still need this TODO?

Reply:
Yes, because the lease store operation does not return any `WriteOp`. This requires a refine to make the code looks more unified, this could be done in the future.